src/StaticAnalysis/PHPStan/Process/PHPStanMutantProcessFactory.php

```diff
@@ -78,18 +86,42 @@ public function create(Mutant $mutant): MutantProcess
     private function getMutantCommandLine(
         string $mutatedFilePath,
         string $mutationOriginalFilePath,
+        string $mutantConfigFile,
     ): array {
         return $this->commandLineBuilder->build(
             $this->staticAnalysisToolExecutable,
             [],
             [
                 "--tmp-file=$mutatedFilePath",
                 "--instead-of=$mutationOriginalFilePath",
+                "--configuration=$mutantConfigFile",
                 '--error-format=json',
                 '--no-progress',
                 '-vv',
                 // todo [phpstan-integration] --stop-on-first-error
             ],
         );
     }
+
+    private function buildMutationConfigFile(string $mutationHash): string
+    {
+        $mutantConfigPath = sprintf(
+            '%s/phpstan.%s.infection.neon',
+            $this->tmpDir,
+            $mutationHash,
+        );
+
+        $this->fileSystem->dumpFile(
```

Comment:
not a blocker for here by any means, but it would be great if this could be passed via the CLI instead as it otherwise means we need to dump one more file for every mutation, making it yet more IO intensive

Reply:
I realise it is not something we'll be able to do every time neither, but it would be idea if we could have one main config dumped (if necessary) for the whole mutation run and then having the mutator specific config configured via CLI parameters

Reply:
its worth trying. but you need to make sure PHPStan can still utlize the result cache.
depending on which parameters you pass or which settings you change in the config, it might decide it need to re-build its result cache

Reply:
btw: there is a option you can pass to make PHPStan fail it it was not able to use the result cache.
https://github.com/phpstan/phpstan-src/blob/6095601023ab1fe4bc9371c09679aaf979e02152/src/Command/AnalyseCommand.php#L111

this might be useful that we don't accidentally do a change and make the result cache useless